mysql/datadog_checks/mysql/mysql.py

```diff
@@ -216,12 +213,21 @@ def _get_connection_args(self):
             connection_args.update({'port': self._config.port})
         return connection_args
 
-    @contextmanager
-    def _connect(self):
+    def _service_check_tags(self, server=None):
+        # type: (Optional[str]) -> List[str]
+        if server is None:
+            server = self._config.mysql_sock if self._config.mysql_sock != '' else self._config.host
         service_check_tags = [
-            'server:{0}'.format((self._config.mysql_sock if self._config.mysql_sock != '' else self._config.host)),
+            'mysql_server:{0}'.format(server),
```

Comment:
Same comment as with postgres: if we're already making a breaking change with `server` then better drop it entirely in favor of using the correct host tag on all metrics via [resolved hostname](https://github.com/DataDog/integrations-core/blob/3bf453a3a3021a3561bb469e2bdc53345c74ed08/mysql/datadog_checks/mysql/mysql.py#L110-L113).